src/helpers/crawl-cleanup-helper.php

```diff
@@ -0,0 +1,311 @@
+			return \get_permalink( $GLOBALS['post']->ID );
+		}
+		return '';
+	}
+
+	/**
+	 * Returns the proper URL for 404 page.
+	 *
+	 * @param string $current_url The current URL.
+	 * @return string The proper URL.
+	 */
+	public function page_not_found_url( $current_url ) {
+		if ( ! \is_multisite() || \is_subdomain_install() || ! \is_main_site() ) {
+			return '';
+		}
+
+		if ( $current_url !== \home_url() . '/blog/' && $current_url !== \home_url() . '/blog' ) {
+			return '';
+		}
+
+		if ( $this->current_page_helper->is_home_static_page() ) {
+			return \get_permalink( \get_option( 'page_for_posts' ) );
+		}
+
+		return \home_url();
+	}
+
+	/**
+	 * Returns the proper URL for taxonomy page.
+	 *
+	 * @return string The proper URL.
+	 */
+	public function taxonomy_url() {
+		global $wp_query;
+		$term = $wp_query->get_queried_object();
+
+		if ( \is_feed() ) {
+			return \get_term_feed_link( $term->term_id, $term->taxonomy );
+		}
+		return \get_term_link( $term, $term->taxonomy );
+	}
+
+	/**
+	 * Returns the proper URL for search page.
+	 *
+	 * @return string The proper URL.
+	 */
+	public function search_url() {
+		$s = \get_search_query();
+		return \home_url() . '/?s=' . \rawurlencode( $s );
+	}
+
+	/**
+	 * Returns the proper URL for url with page param.
+	 *
+	 * @param string $proper_url The proper URL.
+	 * @return string The proper URL.
+	 */
+	public function query_var_page_url( $proper_url ) {
+		global $wp_query;
+		if ( \is_search( $proper_url ) ) {
+			return \home_url() . '/page/' . $wp_query->query_vars['paged'] . '/?s=' . \rawurlencode( \get_search_query() );
+		}
+		else {
+			return \user_trailingslashit( \trailingslashit( $proper_url ) . 'page/' . $wp_query->query_vars['paged'] );
+		}
+	}
+
+	/**
+	 * Returns true if query is with page param.
+	 *
+	 * @param string $proper_url The proper URL.
+	 * @return bool is query with page param.
+	 */
+	public function is_query_var_page( $proper_url ) {
+		global $wp_query;
+		if ( empty( $proper_url ) || $wp_query->query_vars['paged'] === 0 || $wp_query->post_count === 0 ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Redirects clean permalink.
+	 *
+	 * @param string $proper_url The proper URL.
+	 * @return void.
+	 */
+	public function do_clean_redirect( $proper_url ) {
+		$this->redirect_helper->set_header( 'Content-Type: redirect', true );
+		$this->redirect_helper->remove_header( 'Content-Type' );
+		$this->redirect_helper->remove_header( 'Last-Modified' );
+		$this->redirect_helper->remove_header( 'X-Pingback' );
+
+			$message = \sprintf(
+				/* translators: %1$s: Yoast SEO */
+				\__( '%1$s: unregistered URL parameter removed. See %2$s', 'wordpress-seo' ),
+				'Yoast SEO',
+				'https://yoa.st/advanced-crawl-settings'
+			);
+
+			$this->redirect_helper->do_safe_redirect( $proper_url, 301, $message );
+	}
+
+	/**
+	 * Gets the type of URL.
+	 *
+	 * @return string The type of URL.
+	 */
+	public function get_url_type() {
```

Comment:
I am wondering if this would make more sense in the URL helper.

Reply:
I'm not sure about it because there other types that are not check in this function and also the return value matched the appropriate function in the helper.

Reply:
Makes total sense, that would be really confusing!